spec/preloading_spec.cr

```diff
@@ -2,6 +2,31 @@ require "./spec_helper"
 
 include LazyLoadHelpers
 
+# NOTE: This is only for testing if this is called during a query
+# TODO: Remove once a proper mocking shard is built
+module QueryCalledPatch
+  macro included
+    class_property times_called : Int32 = 0
+
+    def database : Avram::Database.class
+      self.class.times_called += 1
+      previous_def
+    end
+  end
+end
+
+class Comment::BaseQuery
+  include QueryCalledPatch
+end
+
+class SignInCredential::BaseQuery
+  include QueryCalledPatch
+end
+
+class Post::BaseQuery
+  include QueryCalledPatch
+end
```

Comment:
Thoughts on this?

```suggestion
module QuerySpy
  macro included
    class_property times_called : Int32 = 0

    def database : Avram::Database.class
      self.class.times_called += 1
      previous_def
    end
  end
end

class Comment::BaseQuery
  include QuerySpy
end

class SignInCredential::BaseQuery
  include QuerySpy
end

class Post::BaseQuery
  include QuerySpy
end
```

This turned out super well. I love the module abstraction 👍